torch/csrc/jit/codegen/cuda/kernel.cpp

```diff
@@ -33,10 +35,17 @@ class KernelIrScanner : private OptOutDispatch {
 
  private:
   void handle(Expr* expr) final {
+    ++expr_cnt;
     TORCH_CHECK(primary_expressions.insert(expr).second);
     OptOutDispatch::handle(expr);
   }
 
+  void handle(kir::Sync* sync) final {
+    if (sync->isWarHazardSync()) {
+      war_hazard_syncs[expr_cnt] = sync;
```

Comment:
As I mentioned earlier, this is ok for now, but please add a TODO comment to revisit this (and move it in a dedicated validation pass which is not on the common execution/compilation path)

Reply:
:+1: